contracts/payment_splitter/src/lib.rs

```diff
@@ -0,0 +1,91 @@
+//! Payment splitter contract
+//!
+
+#![no_std]
+
+use soroban_sdk::{
+    contract, contracterror, contractimpl, contracttype, panic_with_error, token, Address,
+    ConversionError, Env, Vec,
+};
+
+#[contracterror]
+#[derive(Clone, Debug, Copy, Eq, PartialEq, PartialOrd, Ord)]
+#[repr(u32)]
+pub enum Error {
+    AlreadyInitialized = 1,
+    Conversion = 2,
+    KeyExpected = 3,
+    Overflow = 4,
+}
+
+impl From<ConversionError> for Error {
+    fn from(_: ConversionError) -> Self {
+        Error::Conversion
+    }
+}
+
+#[contracttype]
+#[derive(Clone, Copy)]
+pub enum DataKey {
+    AlreadyInitialized = 0,
+    Admin = 1,
+    Token = 2,
+    PaymentSplit = 3,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, PartialEq)]
+pub struct PaymentSplit {
+    stakeholders: Vec<Address>,
+}
+
+#[contract]
+pub struct PaymentSplitterContract;
+
+#[contractimpl]
+impl PaymentSplitterContract {
+    pub fn init(
+        env: Env,
+        admin: Address,
+        token: Address,
+        stakeholders: Vec<Address>,
```

Comment:
Ensure this is not empty.
Also shouldnt it be possible to push more addresses?

Reply:
hmm good question. i would say that yes, so we avoid redeploying the contract if the membership changes. That would also mean we need to allow removing addresses ? :thinking:

Reply:
I think someone should deploy a new contract for sanity sake.